Add `impersonate` feature to API `/v1/chat/completions`

Yiximail opened the pull request:
The feature same as the impersonate feature on webui.

Add an `impersonate` param to active this feature in `/v1/chat/completions`.

Updated:
Prepend `user_input` in impersonate reply for consistency
It will always returns the entire message now

## Checklist:

- [x] I have read the [Contributing guidelines](https://github.com/oobabooga/text-generation-webui/wiki/Contributing-guidelines).

<details>
<summary> Test Images </summary>

Impersonate:
![image](https://github.com/user-attachments/assets/cf4d7940-cdc3-4dc9-b9c4-3828ef728628)

Normal response:
![image](https://github.com/user-attachments/assets/9e8db58a-f0ab-4948-93ff-64fe77d0a893)

</details>

Yiximail commented:
Is there something wrong with it...? You can tell me, and I can fix it
Sorry for the at @oobabooga 

I use this quite often in api.
I just updated the webui today and realized that it wasn't merged.

jianmomo commented:
> 有什么问题吗……？你可以告诉我，我可以修复它 抱歉提及@oobabooga
> 
> 我在 api 中经常使用它。 我今天刚刚更新了 webui，发现它没有合并。
这个功能是我需要的，但是我好像没有办法拉取这个功能，这太可惜了，请问作者您方便分享下这几个文件吗？

Yiximail commented:
> > Is there something wrong with it...? You can tell me, and I can fix it Sorry for the mention @oobabooga
> > I use this quite often in api. I just updated the webui today and realized that it wasn't merged.
> > I'm not sure what issues there might be with my modifications.
> > I directly copied the contents of typing.py and completions.py, and added a parameter for mimicry in the request. However, there's a problem with the responses—they always start with "ai...".
> 
> <img alt="image" width="1087" src="https://private-user-images.githubusercontent.com/9784092/391001564-178afea7-f1cf-472c-84bb-218f38e65416.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI5MDUzODgsIm5iZiI6MTczMjkwNTA4OCwicGF0aCI6Ii85Nzg0MDkyLzM5MTAwMTU2NC0xNzhhZmVhNy1mMWNmLTQ3MmMtODRiYi0yMThmMzhlNjU0MTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjlUMTgzMTI4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTIyMDcwMDRkMGFjZDU0N2YyNTM0MzNiOGQzMDE2ODEwNDIwNjNhODEzNGRmYjA4NDc1MjI1MGIxNTI0Y2QyYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.G6SgZHjhVNBD6PRtfz3CseiW4D_Z_tT_5JIUnQMPKYM">

~~咳咳，这图不合适吧~~

请提供一下你的模型，和完整参数（带上seed的）

一般这个commit不影响生成内容才对，只是处理了一下prompt的结构

jianmomo commented:
OK, thank you. It seems that I need to learn the framework code carefully. Thank you very much for your help.
